### ISSUE TITLE
Deauth doesn’t work macOS 2021 MacBook Pro 16inch 

### ISSUE THREAD

**Issue by ShadowRoi** [cyrus-and/zizzania#13]
As the title says. Everything else works fine, if I can provide more info please let me know 

Also running macOS Monterey but shouldn’t matter

**Comment by cyrus-and** [comment]
Have you read #6 and #9?

**Comment by ShadowRoi** [comment]
```  Software Versions:
  CoreWLAN:	16.0 (1657)
  CoreWLANKit:	16.0 (1657)
  Menu Extra:	17.0 (1728)
  System Information:	15.0 (1502)
  IO80211 Family:	12.0 (1200.12.2b2)
  Diagnostics:	11.0 (1163)
  AirPort Utility:	6.3.9 (639.16)
  Interfaces:
en0:
  Card Type:	Wi-Fi
  Firmware Version:	wl0: Apr  6 2022 05:55:54 version 20.90.45.0.8.7.118 FWID 01-e7138ff2
  MAC Address:	98:dd:60:57:23:08
  Locale:	ETSI
  Country Code:	GR
  Supported PHY Modes:	802.11 a/b/g/n/ac/ax
  Supported Channels:	1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 36, 40, 44, 48, 52, 56, 60, 64, 100, 104, 108, 112, 116, 120, 124, 128, 132, 136, 140, 149, 153, 157, 161, 165
  Wake On Wireless:	Supported
  AirDrop:	Supported
  AirDrop Channel:	44
  Auto Unlock:	Supported
  Status:	Connected 
```
  
It seems that my Wifi card is different than the 1 from the people that reported the issues. Also I never had some weird shutdown or crashes etc, the only problem is about Deauth. 

So this seems to be more like a hardware problem than software? Apple decided to just remove packet injection? Awesome we are evolving backwards thanks Apple.

**Comment by cyrus-and** [comment]
I don't even know where to start to fix this. I use libpcap for all the low level NIC access. Also IIRC even bettercap had those issues, but it's long time since I checked.

**Comment by ShadowRoi** [comment]
I used bettercap and same things, well worse in fact, there were thousand of errors to just to get it to work and then there was much anomaly as well. Deauth was broken there as well but also it was grabbing handshakes sometimes from networks that were super far away and then using air crack ng was saying that they have no EAPOL data.

With zizzania I did have success as this is the only script that works the well where in my testing it successfully grabbed a handshake and used air crack ng with a wordlist cracked my Wifi password, it grabbed a handshake in the 2-3 try of reconnecting my phone I don’t know if this is a scenario that it grabbed corrupted packets and didn’t get it the first time??

About using a VM with external adapter yes I wish it would work smoothly and great and be done with this but it doesn’t, You see on Apple Silicon we can only run ARM VMs so an ARM version of Kali which I don’t know if this causes  any problems with these tools.
I use a Netgear A7000 I installed drivers that are made from users because there is no official support for Linux and I get all sort of problems, sometimes after some scanning the drivers dies and needs to be replugged, Airodump and other stuff worked which means that monitor mode works and a another test that I did showed that injection also worked with the external USB dongle, and then I tried out zizzania with it and throws me errors that monitor mode isn’t supported… Honestly I don’t have enough strength to debug all of these idk there are errors on all sort of scenarios… Are the drivers faulty?Is it because it’s arm version of Kali?
I don’t even know anymore what did you have success with?
